### PR TITLE
OTT-415: ifa_type for VAST bidder

### PR DESCRIPTION
**Pull request by ShriprasadM**
# Description

https://inside.pubmatic.com:9443/jira/browse/OTT-415

This change addressing following
1. It accepts the valid ifa_type parameter and pass on its to VAST Bidder, if VAST Bidders Tag expects it
2. The macro processor of VAST bidder will replace one or more occurrences of `{ifa_type}` macro that may present in VAST Tag's URL
3. If value is not present for `ifa_type` then macro will be replaced with empty string

# Checklist:

- [ ] PR commit list is unique (rebase/pull with the origin branch to keep master clean).
- [x] JIRA number is added in the PR title and the commit message.
- [ ] Updated the `header-bidding` repo with appropiate commit id.
- [ ] Documented the new changes.

For Prebid upgrade, refer: https://inside.pubmatic.com:8443/confluence/display/Products/Prebid-server+upgrade

### CODE REVIEW

**Review by Pubmatic-Dhruv-Sonone — APPROVED** [review]

**Review by Pubmatic-Dhruv-Sonone — APPROVED** [review]
Please fix the failing checks

**Review by pm-jaydeep-mohite — APPROVED** [review]

**Comment by ShriprasadM** [comment]
> Please fix the failing checks

I don't think its related to this PR change. But after digging into the issue found that the reference we are using `/home/http/GO_SERVER/dmhbserver/` is invalid for validate flow. I have created separate ticket https://inside.pubmatic.com:9443/jira/browse/OTT-453 to fix the validate issue.

Going ahead with ci merge. I am not merging this PR now .. as it is created against master. will create another PR w.r.t CI branch

File: openrtb_ext/device.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ type ExtDevice struct {
 	// Description:
 	//   Prebid extensions for the Device object.
 	Prebid ExtDevicePrebid `json:"prebid"`
+
+	IFAType string `json:"ifa_type"`
```
[the remark this answers is not in the document]
Comment (ShriprasadM) [reply openrtb_ext/device.go:38]: I intentionally not added that to make it inline with other attributes of `ExtDevice`.  Also, I think, it is not used by header-bidding for marshalling/ unmarshalling.

[the remark this answers is not in the document]
Comment (ShriprasadM) [reply openrtb_ext/device.go]: Done the required changes, please review

**Review by pm-viral-vala — APPROVED** [review]
LGTM

**Pull request merged**